flake-modules/dev/default.nix

```diff
@@ -1,17 +1,21 @@
 { inputs, ... }:
 {
-  imports = [
-    inputs.git-hooks.flakeModule
-    inputs.treefmt-nix.flakeModule
-    ./devshell.nix
-  ];
+  imports =
+    [ ./devshell.nix ]
+    ++ (if inputs.git-hooks ? flakeModule then [ inputs.git-hooks.flakeModule ] else [ ])
+    ++ (if inputs.treefmt-nix ? flakeModule then [ inputs.treefmt-nix.flakeModule ] else [ ]);
```

Comment:
Couldn't we use `lib.optional` here ?

Reply:
looks like we can. i assumed it would lead to infinite recursion with the module argument and `imports` nonsense you'd usually get in nixos configurations 😆

Reply:
`lib.optional` is implemented using `if then else`, so it does the same thing, just prettier